setup_dependencies.sh

```diff
@@ -56,103 +25,144 @@ function detect_distribution ()
+    local message=$2
+
+    if [ ${result} -ne 0 ]
     then
-        $(dirname "$0")/modules/open-cas-telemetry-framework/setup_dependencies.sh
-        check_result $? "Could not install dependencies"
-    else
-        info "No OCTF in the project source tree, assuming OCTF and dependencies are installed systemwide"
+        printf "[IOTRACE][ERROR] ${message}\n" 1>&2
+        exit ${result}
     fi
 }
 
+#
+# Usage: iotrace_error <ERROR_MESSAGE_1> [ <ERROR_MESSAGE_2> ... ]
+# Note: exits with error
+#
+function iotrace_error () {
+    iotrace_check_result 255 "$*"
+}
 
-if [ "$EUID" -ne 0 ]
-then
-    echo "Please run as root to alllow using apt/yum and installing to /opt"
-    exit 1
-fi
+#
+# Usage: iotrace_info <INFO_MESSAGE_1> [ <INFO_MESSAGE_2> ... ]
+#
+function iotrace_info () {
+    echo "[IOTRACE][INFO] $*" 1>&2
+}
 
-distro=$(detect_distribution)
-case "${distro}" in
-"RHEL7")
-    info "RHEL7.x detected"
-    packages="kernel-devel-$(uname -r)"
+function iotrace_get_kernel_package () {
+    distro=$(detect_distribution)
+    case "${distro}" in
+    "RHEL7"|"CENTOS7")
+        echo "kernel-devel"
+        ;;
+    "FEDORA")
+        echo "kernel-devel"
+        ;;
+    "UBUNTU")
+        echo "linux-headers"
+        ;;
+    *)
+        iotrace_error "Unknown Linux distribution"
+        ;;
+    esac
+}
 
-    info "Installing packages: ${packages}"
-    yum -y install ${packages}
-    check_result $? "Cannot install required dependencies"
+function iotrace_get_distribution_pkg_depndenices () {
+    distro=$(detect_distribution)
+    case "${distro}" in
+    "RHEL7"|"CENTOS7"|"FEDORA"|"UBUNTU")
+        echo "rpm-build dpkg"
+        ;;
+    *)
+        iotrace_error "Unknown Linux distribution"
+        ;;
+    esac
+}
 
-    if [ ! -d /usr/src/kernels/$(uname -r) ]
+function iotrace_setup_kernel_headers () {
+    local kernel_version=$(uname -r)
+    local kernel_dir=$(readlink -f /lib/modules/${kernel_version}/build)
+
+    if [ -d "${kernel_dir}" ]
     then
-        error "Linux kernel headers were not properly installed."
+        iotrace_info "Kernel headers already available: ${kernel_dir}"
+        return 0
     fi
 
-    setup_other_deps
-    ;;
-"CENTOS7")
-    info "CentOS7.x detected"
-    packages="kernel-devel-$(uname -r)"
-
-    info "Installing packages: ${packages}"
-    yum -y install ${packages}
-    check_result $? "Cannot install required dependencies"
+    iotrace_info "Install kernel headers"
+    ${PKG_INSTALLER} $(iotrace_get_kernel_package)
+    iotrace_check_result $? "Cannot install kernel headers"
 
-    if [ ! -d /usr/src/kernels/$(uname -r) ]
+    kernel_dir=$(readlink -f /lib/modules/${kernel_version}/build)
+    if [ ! -d "${kernel_dir}" ]
     then
-        error "Linux kernel headers were not properly installed."
+        iotrace_info "Please run the kernel appropriate to the kernel headers"
+        iotrace_error "Installed kernel headers do not match to the running kernel version."
```

Comment:
Maybe let's add a note "Consider updating system kernel to match available headers" to suggest next action.